### PR TITLE
Replace hardcoded python3 by python{pyv}

### PR DESCRIPTION
**Pull request by mfeurer**
I observed that python3 does not work in my singularity build as it is not a symlink to the system python, but something else:

```
-rwxr-xr-x 1 root root 5482296 Jul  2 07:54 python
-rwxr-xr-x 1 root root 5482296 Jul  2 07:54 python3
lrwxrwxrwx 1 root root      18 Jul  2 07:56 python3.7 -> /usr/bin/python3.7
```

I therefore propose to use the python version that's given by the `{pyv}` variable everywhere, also in the docker file.

In addition, this PR adds some brackets to the singularity file creation and removes a `RUN` which broke my build.

### CODE REVIEW

**Comment by mfeurer** [comment]
CC @franchuterivera

File: amlb/runners/singularity.py [hunk 1]
```diff
@@ -187,9 +186,10 @@ def _generate_script(self, custom_commands):
 # packages that we need to data pre- and postprocessing without breaking it.
 $SPIP install -U pip wheel
 $SPY -m venv venv
-PIP=/bench/venv/bin/python3 -m pip
-PY=/bench/venv/bin/python3
+PIP="/bench/venv/bin/python{pyv} -m pip"
+PY="/bench/venv/bin/python{pyv} -W ignore"
 #$PIP install -U pip=={pipv} wheel
+ls -al /bench/venv/bin/
```
Comment (PGijsbers) [inline amlb/runners/singularity.py:192]:
```suggestion
```

**Comment by sebhrusen** [comment]
@mfeurer is it a specificity of singularity?
This is what I get in a existing docker image:
```
lrwxrwxrwx 1 root root    9 Apr 29 02:53 python -> python3.7
lrwxrwxrwx 1 root root    9 Apr 29 02:53 python3 -> python3.7
lrwxrwxrwx 1 root root   18 Apr 29 02:53 python3.7 -> /usr/bin/python3.7
```

**Review by sebhrusen — APPROVED** [review]

**Comment by mfeurer** [comment]
Yes, this is specific to singularity. Docker works fine. Unfortunately, I don't know what causes this difference in behavior.

**Pull request merged**